helpers/errorHandlers.js

```diff
@@ -2,38 +2,60 @@ const logger = require('winston');
 const { postToStats } = require('../controllers/statsController.js');
 
 module.exports = {
-  handleRequestError (action, originalUrl, ip, error, res) {
-    logger.error(`Request Error: ${originalUrl}`, module.exports.useObjectPropertiesIfNoKeys(error));
-    postToStats(action, originalUrl, ip, null, null, error);
-    if (error.response) {
-      res.status(error.response.status).render('requestError', {message: error.response.data.error.message, status: error.response.status});
-    } else if (error.code === 'ECONNREFUSED') {
-      res.status(503).render('requestError', {message: 'Connection refused.  The daemon may not be running.', status: 503});
-    } else if (error.message) {
-      res.status(500).render('requestError', {message: error.message, status: 500});
-    } else {
-      res.status(400).render('requestError', {message: error, status: 400});
-    }
-  },
-  handlePublishError (error) {
-    logger.error('Publish Error:', module.exports.useObjectPropertiesIfNoKeys(error));
+  returnErrorMessageAndStatus: function (error) {
+    let status, message;
+    // check for daemon being turned off
     if (error.code === 'ECONNREFUSED') {
-      return 'Connection refused.  The daemon may not be running.';
+      status = 503;
+      message = 'Connection refused.  The daemon may not be running.';
+    // check for errors from the deamon
     } else if (error.response) {
+      status = error.response.status || 500;
       if (error.response.data) {
         if (error.response.data.message) {
-          return error.response.data.message;
+          message = error.response.data.message;
         } else if (error.response.data.error) {
-          return error.response.data.error.message;
+          message = error.response.data.error.message;
+        } else {
+          message = error.response.data;
         }
-        return error.response.data;
+      } else {
+        message = error.response;
       }
-      return error.response;
+    // check for spee.ch thrown errors
+    } else if (error.message) {
+      status = 400;
+      message = error.message;
+    // fallback for everything else
     } else {
-      return error;
+      status = 400;
+      message = error;
     }
+    return [status, message];
+  },
+  handleRequestError: function (action, originalUrl, ip, error, res) {
+    logger.error(`Request Error on ${originalUrl}`, module.exports.useObjectPropertiesIfNoKeys(error));
+    postToStats(action, originalUrl, ip, null, null, error);
+    const errorStatusAndMessage = this.returnErrorMessageAndStatus(error);
```

Comment:
[error, message] = this.returnErrorMessageAndStatus(error)

Prob worth spending some time reading/learning on ES2016 destructuring as well as spread operators.